### PR TITLE
Task2

### PR DESCRIPTION
**Pull request by kkulahin**
# React task 2 

### Task: 
Create two additional components and render them in <Main /> component. One component should be written as a class, and should have state and proper lifecycle methods (e.g. <CardsContainer />). This component should fetch the data, update its own state and then map this state to a presentational child component (e.g. <Card />). 

 

**_Steps which should be covered:_**

1) Create an additional folder in your project’s root directory which will be called __api__. 

This folded should contain 2 files: __mockedResponse.js__, and __mockedApi.js__ 

files

2) __mockedResponse.js__ file should just contain one variable, which will be an array of objects (make 6-8 objects with optional fields). For example:

files
 

3) the __mockedApi.js__ file contains function, which will return a resolved promise with __mockedResponse__ data 

 files

4) in your project’s __‘src’__ folder create an additional folder called __‘components’__ and move all components there (you imports might need to be updated accordingly). 

The result should be something similar

files

 

5) In your __components__ folder create an additional folder which will be called __Cards__. 

__This folder should contain 2 files for 2 components: `<CardsContainer/>` and `<Card />` and related CSS-files for each of them.__

Note: `<CardsContainer />` is a class component and it should have state and proper lifecycle methods. `<Card />` component is just a pure functional component which just should accept several props and render them in a simple card with image and description (design of the card is up to you) 

6) `<CardsContainer/>` should fetch the data from our mockedApi and update it’s own state. After it the actual state should be mapped to a `<Card />` component. (don’t forget about key property)  

files
 

7) ___If there is no cards yet render a fallback JSX___ like `<div>No cards yet</div>`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    "stylelint": "npx stylelint src/**/*.s{c,a}ss",
+    "lint:fix": "tsc --noEmit && eslint --fix src/**/*.{js,ts,tsx} --no-error-on-unmatched-pattern",
+    "stylelint:fix": "npx stylelint --fix src/**/*.s{c,a}ss",
+    "lint:all": "npm run lint:fix && npm run stylelint:fix",
+    "start:dev": "webpack serve --config=configs/webpack/dev.js --env NODE_ENV=development --env BUILD_TYPE=local",
+    "start:prod": "npm run build && npm start",
+    "start": "node express.js"
+  },
+  "devDependencies": {
+    "@babel/cli": "^7.13.14",
```
Comment (MishkanJigan) [inline package.json]: Remove dynamic dependencies will be better.

File: src/components/Cards/CardsContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  }
+
+  updateCardsState = (cards: ICard[]) => this.setState({ cards });
+
+  renderCards = () => {
+    const { cards } = this.state;
+    if (!cards || !cards.length) {
+      return <div>No cards yet</div>;
+    }
+    return cards.map(({ gender, id, imageUrl, price, title }) => {
```
Comment (MishkanJigan) [inline src/components/Cards/CardsContainer.tsx]: What do you think about make it this way and then use card={card}?
```suggestion
    return cards.map((card: ICard) => {
```

File: package.json (hunk 1 above)
Comment (kkulahin) [reply package.json]: FIxed

File: src/components/Cards/CardsContainer.tsx (hunk 2 above)
Comment (kkulahin) [reply src/components/Cards/CardsContainer.tsx]: Fixed
https://github.com/BlackTruth/react/blob/5e20841788b32724609b7a987dbab75974a9c0d3/src/components/Cards/CardsContainer.tsx#L33-L37

**Pull request merged**